File: c++/C++ Part2/inheritance/inheritCode.cpp
//Program to demonstrate : 
//Inheritance of code
//refer : inheritance.txt
//see : inheritCode.png

#include <iostream>
using namespace std;

class A
{
public:
    void f1();
    void f2();
};

void A :: f1()
{
    cout<<"\n A f1()";
}

void A :: f2()
{
    cout<<"\n A f2()";
}

class B : public A
{
public:
    void f2();
    void f3();
    
};

void B :: f2()
{
    cout<<"\n B f2()";
}
void B :: f3()
{
    cout<<"\n B f3()";
}

int main()
{
   B objB; 
   objB.f1();//inherited code
   objB.f2();//overridden code
   objB.f3();//extended code
   
   return 0;
}

File: Java/Inheritance/InheritCode.java
class A
{
 void f1()
 {
  System.out.println("A f1()");
 }

 void f2()
 {
  System.out.println("A f2()");
 }
}//A

class B extends A
{
 //redefines an inherited method
 //i.e. overrides
 void f2()
 {
  System.out.println("B f2() starts");
  f3(); //connect with the extended code
  System.out.println("B f2() ends");
 }

 void f3()
 {
  System.out.println("B f3()");
 }
}

class InheritCode
{
 //execution
 public static void main(String args[])
 {
  B objB = new B();
  objB.f1();//inherited code
  objB.f2();//overridden code
 }//main
}//InheritCode